src/syscheckd/src/db/testtool/main.cpp

```diff
@@ -0,0 +1,127 @@
+}
+
+int main(int argc, const char* argv[])
+{
+    try
+    {
+        CmdLineArgs cmdLineArgs(argc, argv);
+
+        const auto actions { cmdLineArgs.actions() };
+
+        std::cout << "Actions: " << actions.size() << std::endl;
+
+        std::ifstream configFile{ cmdLineArgs.configFile() };
+        if (configFile.good())
+        {
+            const auto& jsonConfigFile { nlohmann::json::parse(configFile) };
+            const auto storageType{ jsonConfigFile.at("storage_type").get<const uint32_t>() };
+            const auto syncInterval{ jsonConfigFile.at("sync_interval").get<const uint32_t>() };
+            const auto fileLimit{ jsonConfigFile.at("file_limit").get<const uint32_t>() };
+            const auto valueLimit{ jsonConfigFile.at("value_limit").get<const uint32_t>() };
+            const auto isWindows{ jsonConfigFile.at("is_windows").get<const bool>() };
+
+            std::function<void(const std::string&)> callbackSyncFileWrapper
+            {
+                [](const std::string & msg)
+                {
+                    syncCallback(FIM_COMPONENT_FILE, msg.c_str());
+                }
+            };
+
+            std::function<void(const std::string&)> callbackSyncRegistryWrapper
+            {
+                [](const std::string & msg)
+                {
+                    syncCallback(FIM_COMPONENT_REGISTRY, msg.c_str());
+                }
+            };
+
+            std::function<void(modules_log_level_t, const std::string&)> callbackLogWrapper
+            {
+                [](modules_log_level_t level, const std::string & log)
+                {
+                    loggerFunction(level, log.c_str());
+                }
+            };
+
+            try
+            {
+                DB::instance().init(storageType,
+                    syncInterval,
+                    callbackSyncFileWrapper,
+                    callbackSyncRegistryWrapper,
+                    callbackLogWrapper,
+                    fileLimit,
+                    valueLimit,
+                    isWindows);
+
+                std::unique_ptr<TestContext> testContext { std::make_unique<TestContext>()};
+                testContext->outputPath = cmdLineArgs.outputFolder();
+
+                // Let's take the input json list and apply the changes to the db
+                for (size_t idx = 0; idx < actions.size(); ++idx)
+                {
+                    testContext->currentId = idx;
+                    const std::string inputFile{ actions[idx] };
+                    std::ifstream actionsIdxFile{ inputFile };
+                    std::cout << "Processing file: " << inputFile << std::endl;
+                    const auto& jsonAction { nlohmann::json::parse(actionsIdxFile) };
+                    auto action { FactoryAction::create(jsonAction.at("action").get<std::string>()) };
+                    action->execute(testContext, jsonAction.at("body"));
+                }
+
+                //DB::stopIntegrity();
```

Comment:
Here we can add a call to "DB::teardown()" in order to close correctly the database and synchronization